### ISSUE TITLE
Server upgrade required warning message appears despite installed version 5.31

### ISSUE THREAD

**Issue by m-a-v** [mattermost/mattermost-mobile#5338]
#### Summary
Server upgrade required warning message appears despite installed version 5.31.

#### Environment Information

- Device Name: iPhone XS Max
- OS Version: iOS 14.4.2
- Mattermost App Version: 1.42.0 (Build 353) iOS
- Mattermost Server Version: 5.31.1

#### Expected behavior
Warning message should not appear!

#### Observed behavior
A warning message appears, when using the newest mobile app version (iOS and Android).

**Screenshot when starting the iOS app**
![Warning message](https://user-images.githubusercontent.com/16197536/115231594-09298300-a116-11eb-9127-27e73f1ccfae.png)

**Displayed Server Version when using the app**
![App Information](https://user-images.githubusercontent.com/16197536/115231612-0d55a080-a116-11eb-8b4d-3752bb80e5f6.png)

**Server Version when using the web client**
<img width="185" alt="Server Information (using browser)" src="https://user-images.githubusercontent.com/16197536/115231620-10509100-a116-11eb-8ffb-485659469179.png">

@amyblais Could it be that the database schema version is used for version comparison? This would be a bug!

**Comment by rcaillon-Iliad** [comment]
Same problem:
- server: 5.31.2
- client android 1.42.0

**Comment by amyblais** [comment]
@m-a-v @rcaillon-Iliad There was a bug that has been fixed in v5.31.3. Would you be open to upgrading to see if it fixes the issue?

**Comment by rcaillon-Iliad** [comment]
My admin sys is currently on vacation :/

**Comment by amyblais** [comment]
Servers need to be on v5.31.3 because [v5.31.3 fixes an issue where the server version was reported as v5.30.0](https://docs.mattermost.com/administration/changelog.html#release-v5-31-esr).

Mobile app changelog has been updated with the details https://docs.mattermost.com/administration/mobile-changelog.html, and we have pushed a v1.42.1 dot release to the app stores that clarifies the minimum supported version in the in-app notice. It should be available for download soon.
